packages/webpack-cli/lib/utils/arg-utils.js

```diff
@@ -18,6 +21,37 @@ function arrayToObject(arr) {
     }, {});
 }
 
+function resolveFilePath(filename = '', defaultValue) {
+    if (filename && Array.isArray(filename)) {
+        return filename.map((fp) => resolveFilePath(fp, defaultValue)).filter((e) => e);
+    }
+    if (filename && !filename.includes('.js')) {
+        filename = filename + '.js';
+    }
+    if (defaultValue && !defaultValue.includes('.js')) {
+        defaultValue = defaultValue + '.js';
+    }
+    let configPath;
+    const predefinedConfigPath = filename ? resolve(process.cwd(), filename) : undefined;
+    const configPathExists = predefinedConfigPath ? existsSync(predefinedConfigPath) : undefined;
+
+    if (!configPathExists) {
+        const LOOKUP_PATHS = [`${filename}`, `src/${filename}`, ...(defaultValue ? [defaultValue, `src/${defaultValue}`] : [])];
+
+        if (filename) {
+            LOOKUP_PATHS.unshift(filename);
+        }
+        LOOKUP_PATHS.forEach((p) => {
+            const lookUpPath = join(process.cwd(), ...p.split('/'));
+            if (existsSync(lookUpPath) && !configPath) {
+                configPath = lookUpPath;
+            }
+        });
+    }
+    return configPathExists ? predefinedConfigPath : configPath;
```

Comment:
To be honestly it is wrong, webpack uses resolved to resolve `js`/`ts`/`cjs`/`mjs` and other extensions, I think It should be removed

Reply:
I think all test will be passed without it

Reply:
makes sense, let's simply do `path.resolve`

Reply:
Ah so we don't even need to do path.resolve since core does this already for relative paths, we just need to pass the entry